### PR TITLE
Add way to create account with delegate in 1 tx

### PR DESCRIPTION
**Pull request by garious**
#### Problem

Delegating stake currently requires a separate transaction

#### Summary of Changes

Add a 3-instruction transaction to create, init, and delegate in one transaction.

### CODE REVIEW

File: programs/vote_api/src/vote_transaction.rs [hunk 1]
```diff
@@ -30,27 +29,48 @@ impl VoteTransaction {
...
+        TransactionBuilder::new(fee)
+            .push(SystemInstruction::new_program_account(
+                from_id,
+                voter_id,
+                num_tokens,
+                space,
+                id(),
+            ))
+            .push(VoteInstruction::new_initialize_account(voter_id))
+            .push(VoteInstruction::new_delegate_stake(voter_id, delegate_id))
```
Comment (sagar-solana) [inline programs/vote_api/src/vote_transaction.rs]: Question: How does this get past the AccountLoadedTwice error in runtime? 
```
if has_duplicates(&tx.account_keys) {
                error_counters.account_loaded_twice += 1;
                return Err(BankError::AccountLoadedTwice);
            }
```
Is the transactionbuilder doing some magic here?

Comment (garious) [reply programs/vote_api/src/vote_transaction.rs]: Only a single instruction can't reference the same account twice. It's no problem across multiple instructions.

Comment (garious) [reply programs/vote_api/src/vote_transaction.rs]: Of maybe you're talking about this magic. :) https://github.com/solana-labs/solana/blob/master/sdk/src/transaction_builder.rs#L15

Comment (CriesofCarrots) [reply programs/vote_api/src/vote_transaction.rs]: When the same account is used by multiple instructions, it is only added to tx.account_keys once, and identified by the instruction indexes.
The AccountLoadedTwice error is only triggered is the same account is actually added to tx.account_keys twice.

Comment (CriesofCarrots) [reply programs/vote_api/src/vote_transaction.rs]: (Oops, sorry for the overkill. Should have refreshed!)

Comment (sagar-solana) [reply programs/vote_api/src/vote_transaction.rs]: Yup that's how I expected it to work, makes sense now. Thanks!

**Pull request merged**